Documentation on what Telemetry Data is collected + how to opt-in/opt-out (#5253)
nginxinc/kubernetes-ingress: docs/content/overview/product-telemetry.md

---
title: "Product Telemetry"
description: "Learn why NGINX Ingress Controller collects telemetry, and understand how and what it gathers."
weight: 500
toc: true
---

## Overview

NGINX Ingress Controller collects product telemetry data to allow its developers to understand how it's deployed and configured by users.

> This data is used to triage development work, prioritizing features and functionality that will benefit the most people.

Reviewer: ```suggestion
This data is used to triage development work, prioritizing features and functionality that will benefit the most people.

```

Adding a new line for consistency with the other sentences.

Product telemetry is enabled by default, collected once every 24 hours. It's then sent to a service managed by F5 over HTTPS.

{{< note >}}
If you would prefer to avoid sending any telemetry data, you can [opt-out](#opt-out) when installing NGINX Ingress Controller.
{{< /note >}}

## Data Collected

> These are the data points collected and reported by the NGINX Ingress Controller:

Reviewer: ```suggestion
These are the data points collected and reported by NGINX Ingress Controller:
```

- **Project Name** The name of the software, which will be labelled `NIC`.
- **Project Version** The NGINX Ingress Controller version.
- **Project Version** The NGINX Ingress Controller version.
- **Installation ID** Used to identify a unique installation of NGINX Ingress Controller.
- **VirtualServers** The number of VirtualServer resources managed by NGINX Ingress Controller.
- **VirtualServerRoutes** The number of VirtualServerRoute resources managed by NGINX Ingress Controller.
- **TransportServers** The number of TransportServer resources managed by the NGINX Ingress Controller.

## Opt out

Product telemetry can be disabled when installing NGINX Ingress Controller.

### Helm

> When installing or upgrading NGINX Ingress Controller with Helm, set the `controller.telemetry.enable` option to `false`

Reviewer: ```suggestion

When installing or upgrading NGINX Ingress Controller with Helm, set the `controller.telemetry.enable` option to `false`
```

New line for spacing and readability.

---
title: "Product Telemetry"
description: "Learn how and why NGINX Ingress Controller collects product telemetry data."
weight: 500
toc: true
---

## Overview

> The NGINX Ingress Controller collects product telemetry data to allow its developers to better understand the way our users deploy and configure our product.

Reviewer: ```suggestion
NGINX Ingress Controller collects product telemetry data to allow its developers to understand how it's deployed and configured by users.
```

The goal is to use this data to triage development work and prioritize features and functionality that will provide the most value to our users.

## About Product Telemetry

> Product telemetry is collected by default.
> Data is collected once every 24 hours, and is sent to a service managed by F5 over HTTPS.
> Personally identifiable information (PII) is **not** collected. 

Reviewer: ```suggestion
Product telemetry is enabled by default, collected once every 24 hours. It's then sent to a service managed by F5 over HTTPS.
```

Reducing the paragraphs: each sentence is relatively short. It's jarring having PII mentioned because it doesn't feel relevant in this case: it would be a concern for a user-facing application such as a mobile app, not an infrastructure tool.

> > [!NOTE]  
> > If you prefer to not have data collected, you can [opt-out](#opt-out) when installing NGINX Ingress Controller.

Reviewer: ```suggestion
{{< note >}}
If you would prefer to avoid sending any telemetry data, you can [opt-out](#opt-out) when installing NGINX Ingress Controller.
{{< /note >}}
```

This is going into Hugo, so we can use Hugo shortcodes. "prefer to not have" is a double negative and harder to read as a result. Even worse for non-native readers.

## Data Collected

> Below is a list of current set data points that are collected and reported by the NGINX Ingress Controller:

Reviewer: ```suggestion
These are the data points collected and reported by the NGINX Ingress Controller:
```
It's self-evident as a list, so there's no need to mention it's a list or announce a list is about to be shown when the reader is going top down anyway.

"current set data points" is an awkward term. Data points are implicitly precise, and even if there is the possibility of additional points in the future, mentioning it could put people off. If there's no plans, there's no need for the specificity of "current".

> - **Project Name** This is the name of the product. In this case `NIC`

Reviewer: ```suggestion
- **Project Name** The name of the software, which will be labelled `NIC`.
```

NIC is not the name, it's a shorthand acronym.

> - **Project Version** The version of the NGINX Ingress Controller.

Reviewer: ```suggestion
- **Project Version** The NGINX Ingress Controller version.
```

> - **Project Architecture** The architecture of the kubernetes environment. (e.g. amd64, arm64, etc...)
> - **Cluster ID** A unique identifier of the kubernetes cluster that the NGINX Ingress Controller is deployed to.
> - **Cluster Platform** The platform that the kubernetes cluster is operating on. (e.g. eks, aks,  etc...)

Reviewer: ```suggestion
- **Project Architecture** The architecture of the Kubernetes environment. (Such as amd64, arm64, etc...)
- **Cluster ID** A unique identifier of the Kubernetes cluster that NGINX Ingress Controller is deployed to.
- **Cluster Platform** The platform that the Kubernetes cluster is operating on. (Such as eks, aks,  etc...)
```
Kubernetes is a proper noun. We've in the middle of working on a new style guide and have decided we're going to try to avoid Latinisms in how we write documentation.

> - **Installation ID** Used to identify a unique installation of the NGINX Ingress Controller.
> - **VirtualServers** A count of the number of VirtualServer resources managed by the NGINX Ingress Controller.
> - **VirtualServerRoutes** A count of the number of VirtualServerRoute resources managed by the NGINX Ingress Controller.
> - **TransportServers** A count of the number of TransportServer resources managed by the NGINX Ingress Controller.

Reviewer: ```suggestion
- **Installation ID** Used to identify a unique installation of NGINX Ingress Controller.
- **VirtualServers** The number of VirtualServer resources managed by NGINX Ingress Controller.
- **VirtualServerRoutes** The number of VirtualServerRoute resources managed by NGINX Ingress Controller.
- **TransportServers** The number of TransportServer resources managed by the NGINX Ingress Controller.
```

Removing unnecessary words.

## Opt out

> Collection and reporting of product telemetry can be switched off when installing the NGINX Ingress Controller.

Reviewer: ```suggestion
Product telemetry can be disabled when installing NGINX Ingress Controller.
```

### Helm

> When installing/upgrading with Helm, set the `controller.telemetry.enable` option to `false`

Reviewer: ```suggestion
When installing or upgrading NGINX Ingress Controller with Helm, set the `controller.telemetry.enable` option to `false`
```

This can be set directly in the `values.yaml` file, or using the `--set` option

```shell
helm upgrade --install ... --set controller.telemetry.enable=false
```

> ### Manifest
> 
> When installing with Manifest, set the `-enable-telemetry-reporting` flag to `false`

Reviewer: ```suggestion
### Manifests

When installing NGINX Ingress Controller with Manifests, set the `-enable-telemetry-reporting` flag to `false`
```
